# Sherlock-Lung Piolot Study
Bioinformatic analysis scripts for Sherlock-Lung pilot study. We inlcuded four differnt somatic callers genome wide somatic varaint calling. In summary, the analysis-ready BAM files were processed using four different algorithms, including MuTect, MuTect, Strelka, and TNscope. To improve the performance of the variant calling, we used Sentieonâ€™s genomics package to run MuTect, MuTect and TNscope. Only those SNVs that passed calling by a minimum of three algorithms were kept. To reduce false positive calling, we applied this in-house filtering script similar to our previous publication. To summarize, variant calling was considered only at the genome positions with 1) read depth >12 in tumor and >6 in normal samples; and 2) variant read count >5 in tumor and VAF <0.02 in normal samples. The filtered variants were annotated with Oncotator and ANNOVAR. For the indel calling, only variants called by three algorithms were kept (MuTect2, TNscope, and Strelka). UPS-indel algorithm was used to compare and combine different indel call sets. Similar filtering steps as those used for SNV calling were also applied to indel call. We recommend to remove possible germline variants from the called somatic variants, using the dbSNP138, 1000 genomes, ExAC, gnomAD database, or any in-house germline variant database for commonly occurring SNPs (somatic variant frequency<0.001). 
